packages/arb-token-bridge-ui/src/components/TransferPanel/DepositCardL2Failure.tsx

```diff
@@ -19,40 +30,75 @@ export function DepositCardL2Failure({ tx }: { tx: MergedTransaction }) {
     [isConnectedToArbitrum]
   )
 
+  useEffect(() => {
+    ;(async () => {
+      const { daysTillExpiry, isValid } = await getRetryableTicketExpiration({
+        l1TxHash: tx.txId,
+        l1Provider: l1Signer.provider,
+        l2Provider: l2Signer.provider
+      })
+
+      setRetryableExpiryDays({
+        days: daysTillExpiry,
+        isValid
+      })
+    })()
+  }, [l1Signer.provider, l2Signer.provider, tx.txId])
```

Comment:
Instead of doing an IIFE, let's define the function and call it, seems a bit more clear:

```suggestion
useEffect(() => {
  async function updateRetryableTicketExpirationDate() {
    const { daysTillExpiry, isValid } = await getRetryableTicketExpiration({
      l1TxHash: tx.txId,
      l1Provider: l1Signer.provider,
      l2Provider: l2Signer.provider
    })

    setRetryableExpiryDays({
      days: daysTillExpiry,
      isValid
    })
  }
  
  updateRetryableTicketExpirationDate()
}, [l1Signer.provider, l2Signer.provider, tx.txId])
```

Reply:
Done - looks much better.
Added a useCallback as well.